src/web/app/pages-instructor/instructor-session-individual-extension-page/instructor-session-individual-extension-page.component.ts

```diff
@@ -0,0 +1,528 @@
+        updateDeadlinesType, extensionTimestamp),
+      instructorDeadlines: this.getUpdatedDeadlines(selectedInstructors, this.instructorDeadlines,
+        updateDeadlinesType, extensionTimestamp),
+      ...this.feedbackSessionDetails,
+    };
+
+    this.isSubmittingDeadlines = true;
+    this.feedbackSessionsService
+      .updateFeedbackSession(this.courseId, this.feedbackSessionName, request, isNotifyDeadlines)
+      .pipe(finalize(() => { this.isSubmittingDeadlines = false; }))
+      .subscribe(() => {
+          this.loadFeedbackSessionAndIndividuals();
+          this.showSucessToast(updateDeadlinesType, selectedStudents, selectedInstructors);
+        },
+        (resp: ErrorMessageOutput) => {
+          this.statusMessageService.showErrorToast(resp.error.message);
+        },
+      );
+  }
+
+  private showSucessToast(updateDeadlinesType: DeadlineHandlerType,
+    selectedStudents: StudentExtensionTableColumnModel[],
+    selectedInstructors: InstructorExtensionTableColumnModel[],
+  ): void {
+    const updateAction = updateDeadlinesType === DeadlineHandlerType.CREATE ? 'created' : 'deleted';
+    this.statusMessageService.showSuccessToast(
+      `Successfully ${updateAction} extension(s) for ${selectedStudents.length} student(s) and`
+    + ` ${selectedInstructors.length} instructor(s)!`);
+  }
+
+  private getUpdatedDeadlines(
+    selectedIndividuals: StudentExtensionTableColumnModel[] | InstructorExtensionTableColumnModel[],
+    deadlinesToCopyFrom: Record<string, number>,
+    updateDeadlinesType: DeadlineHandlerType,
+    extensionTimestamp?: number,
+  ): Record<string, number> {
+    const record: Record<string, number> = { ...deadlinesToCopyFrom };
+
+    if (updateDeadlinesType === DeadlineHandlerType.CREATE) {
+      selectedIndividuals.forEach((x) => {
+        record[x.email] = extensionTimestamp!;
+      });
+    } else {
+      selectedIndividuals.forEach((x) => {
+        delete record[x.email];
+      });
+    }
+
+    return record;
+  }
+
+  private getSelectedStudents(): StudentExtensionTableColumnModel[] {
+    return this.studentsOfCourse.filter((x) => x.isSelected);
+  }
+
+  getNumberOfSelectedStudents(): number {
+    return this.getSelectedStudents().length;
+  }
+
+  private getSelectedStudentsWithExtensions(): StudentExtensionTableColumnModel[] {
+    return this.studentsOfCourse.filter((x) => x.isSelected && x.hasExtension);
+  }
+
+  private getSelectedInstructors(): InstructorExtensionTableColumnModel[] {
+    return this.instructorsOfCourse.filter((x) => x.isSelected);
+  }
+
+  private getNumberOfSelectedInstructors(): number {
+    return this.getSelectedInstructors().length;
+  }
+
+  private getSelectedInstructorsWithExtensions(): InstructorExtensionTableColumnModel[] {
+    return this.instructorsOfCourse.filter((x) => x.isSelected && x.hasExtension);
+  }
+
+  hasSelected(): boolean {
+    return [...this.studentsOfCourse, ...this.instructorsOfCourse].some((user) => user.isSelected);
```

Comment:
Just being very pedantic: 
Will this be better?
```suggestion
    return this.studentsOfCourse.some((user) => user.isSelected)
        && this.instructorsOfCourse.some((user) => user.isSelected);
```

Reply:
Small mistake in the correction btw. This changes the logic, should be `||` not `&&`.

Reply:
Right.. thanks

Reply:
hence it is being pedantic: the original mandates both `student` and `instructor` to have `isSelected` which is a stronger coupling. Readability wise I feel the same way that the former is more readable.
@samuelfangjw what do you think? A tie-break from you?

Reply:
> the original mandates both student and instructor to have isSelected which is a stronger coupling.

Not sure if I understood this correctly, but I believe they are functionally equivalent? Both are checking if any student or instructor are selected.

No strong opinion on this, but prefer the correction because it better describes the intent in my opinion.